55_revocation_actions_without_python.md

```diff
@@ -0,0 +1,311 @@
+1. Look for the named command in the tenant-provided initial payload
+1. If the agent supports Python actions, look for the Python module on the system
+1. If the agent supports Python actions, look for the Python module in the tenant-provided initial payload
+
+The command takes a command-line argument: the absolute path to the
+file where the revocation message is stored in JSON.  Before being
+invoked, the process' working directory will be changed to the secure
+mount directory (`/var/lib/keylime/secure`) where any initial payload
+is extracted and stored.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+ * A new unit test should be added to exercise action lookup
+ * A new integration test should be written to exercise revocation actions
+ 
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this enhancement
+-->
+
+When downgrading, non-Python actions will stop working.  Proper log
+messages would help diagnose the issues.
+
+### Dependencie requirements
+
+<!--
+If your new change requires new dependencies, please outline and demonstrate that your selected dependency 
+is well maintained and packaged in Keylime's supported Operating Systems (currently Debian Stable
+and as of time writing Fedora 32/33). 
+
+During code implementation you will also be expected to add the package to CI , the keylime ansible role and 
+keylimes main installer (`keylime/installers.sh`).
+
+If the package is not available in the supported Operated systems, the PR will not be merged into master. 
+
+Adding the package in `requirements.txt` is not sufficent for master which is where we tag releases from. 
+
+You may however be able to work within an experimental branch until a package is made available. If this is
+the case, please outline it in this enhancement.
+
+-->
+
+No dependencies are known of.
+
+## Drawbacks
+
+<!--
+Why should this enhancement _not_ be implemented?
+-->
+
+No drawbacks are known of.
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+ * It would also be an option to make use of an embeddable language runtime, such as Lua and mruby.
```

Comment:
In general I like the idea of just having lua scripts for adding new functionality to software.
But in this instance I don't think it is very useful because the revocation script does not really need to interact with the agent itself.